tests/test.sh

```diff
@@ -14,25 +14,65 @@ workflow-install --help
 setup-data-libraries --help
 get-tool-list --help
 
+function start_container {
+    # We start the image with the -P flag that published all exposed container ports
+    # to random free ports on the host, since on OS X the container can't be reached
+    # through the internal network (https://docs.docker.com/docker-for-mac/networking/#i-cannot-ping-my-containers)
+    CID=`docker run -d -e GALAXY_CONFIG_WATCH_TOOL_DATA_DIR=True -P bgruening/galaxy-stable`
+    # We get the webport (https://docs.docker.com/engine/reference/commandline/inspect/#list-all-port-bindings)
+    WEB_PORT=`docker inspect --format="{{(index (index .NetworkSettings.Ports \"$INTERNAL_EXPOSED_WEB_PORT/tcp\") 0).HostPort}}" $CID`
+    echo "Wait for galaxy to start"
+    galaxy-wait -g http://localhost:$WEB_PORT -v --timeout 120
+}
+
+function start_new_container {
+    echo "Start new container"
+    docker rm -f $CID
+    start_container
+}
+
 echo "Starting galaxy docker container"
-# We start the image with the -P flag that published all exposed container ports
-# to random free ports on the host, since on OS X the container can't be reached
-# through the internal network (https://docs.docker.com/docker-for-mac/networking/#i-cannot-ping-my-containers)
-CID=`docker run -d -e GALAXY_CONFIG_WATCH_TOOL_DATA_DIR=True -P bgruening/galaxy-stable`
-# We get the webport (https://docs.docker.com/engine/reference/commandline/inspect/#list-all-port-bindings)
-WEB_PORT=`docker inspect --format="{{(index (index .NetworkSettings.Ports \"$INTERNAL_EXPOSED_WEB_PORT/tcp\") 0).HostPort}}" $CID`
-echo "Test galaxy-wait function"
-galaxy-wait -g http://localhost:$WEB_PORT -v --timeout 120
+start_container
 docker ps
 
-echo "Check tool installation"
+echo "Check tool installation with yaml on the commandline"
+# CD Hit was chosen since it is old and seems to be unmaintained. Last update was 2015.
+# Anyone know a smaller tool that could fit its place?
+OLD_TOOL="{'owner':'jjohnson','name':'cdhit','revisions':['34a799d173f7'],'tool_panel_section_label':'CD_HIT'}"
```

Comment:
We should probably create a dummy tool in the TS. As soon as this container gets updated we will have failing tests that are hard to understand.

Reply:
I agree. Better even to use the test toolshed for this.

Reply:
No, this is fine, the test toolshed is not a good idea. We do similar stuff in galaxy, and even if the repository get's updated this will continue to work.

Reply:
actually we can just make sure that there is a new revision, we don't care which revision that is.

Reply:
@mvdbeek correct me, but if we test the `latest repository` as we do a few lines below, this test will fail as soon as the repo is updated.

Reply:
right, but we can test that there is an additional revision in the tool list

Reply:
> right, but we can test that there is an additional revision in the tool list

Additional revisions are only installed if there are multiple versions. If the revision is of the same version, then it will override the previous revision and not show up with get-tool-list.

CD-HIT was chosen because it was last updated more than 2 years ago. The last update before that was 5 years ago. It seems to be pretty stable. If it changes, it is not too hard to figure out why the tests fail.

If we really want to do this right, we use a docker image to start up a custom toolshed, upload a custom tool to it, and then test it. But I am not sure this is worth the effort. Personally I think the test as it is now is sufficient.